src/ui/storyTellingServerUI.js

```diff
@@ -0,0 +1,174 @@
+    if (!url) { url = window.location.href; }
+    if (name) { name = name.replace(/[\[\]]/g, "\\$&"); }
+    var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+        results = regex.exec(url);
+    if (!results) { return null; }
+    if (!results[2]) { return ""; }
+    return decodeURIComponent(results[2].replace(/\+/g, " "));
+};
+
+/* Loads a story View page and a particular story from a story ID from the query string
+ * - "clientConfig": a collection of client config values consisting of
+ *     - "theme": the current theme of the site
+ *     - "baseTheme": the base theme of the site
+ *     - "authoringEnabled": indicates whether story saving and editing are enabled
+ * - "options": additional options to merge into the View page
+ */
+sjrk.storyTelling.loadStoryFromParameter = function (clientConfig, options) {
+    var storyPromise = fluid.promise();
+
+    var storyId = sjrk.storyTelling.getParameterByName("id");
+
+    if (storyId) {
+        $.get("/stories/" + storyId, function (data) {
+            var retrievedStory = JSON.parse(data);
+
+            options = options || {};
+            options.distributeOptions = [{
+                "target": "{that story}.options.model",
+                "record": retrievedStory
+            },{
+                "target": "{that}.options.pageSetup.authoringEnabled",
+                "record": clientConfig.authoringEnabled
+            }];
+
+            var storyViewComponent = sjrk.storyTelling[clientConfig.theme].page.storyView(options);
+
+            storyPromise.resolve(storyViewComponent);
+        }).fail(function (jqXHR, textStatus, errorThrown) {
+            storyPromise.reject({
+                isError: true,
+                message: errorThrown
+            });
+        });;
+    } else {
+        storyPromise.reject({
+            isError: true,
+            message: "Story not loaded: no story ID provided"
+        });
+    }
+
+    return storyPromise;
+};
+
+/* Loads a story Browse page and populates it with a set of stories
+ * - "clientConfig": a collection of client config values consisting of
+ *     - "theme": the current theme of the site
+ *     - "baseTheme": the base theme of the site
+ *     - "authoringEnabled": indicates whether story saving and editing are enabled
+ * - "options": additional options to merge into the Browse page
+ */
+sjrk.storyTelling.loadBrowse = function (clientConfig, options) {
+    var storiesPromise = fluid.promise();
+
+    $.get("/stories", function (data) {
+        var browseResponse = JSON.parse(data);
+
+        options = options || {};
+        options.distributeOptions = [{
+            "target": "{that storyBrowser}.options.model",
+            "record": browseResponse
+        },{
+            "target": "{that}.options.pageSetup.authoringEnabled",
+            "record": clientConfig.authoringEnabled
+        }];
+
+        var storyBrowseComponent = sjrk.storyTelling[clientConfig.theme].page.storyBrowse(options);
```

Comment:
A big problem with the test set up is that this entire component tree leaks between tests, including the UIO instance nested in it. We need to arrange for teardown of the browse component once the fixtures finish. See below -